internal/cloud/backend_plan.go

```diff
@@ -291,6 +291,27 @@ in order to capture the filesystem context the remote workspace expects:
 			runHeader, b.hostname, b.organization, op.Workspace, r.ID)) + "\n"))
 	}
 
+	// Retrieve the run to get task stages.
+	// Task Stages are calculated upfront so we only need to call this once for the run.
+	taskStages := make([]*tfe.TaskStage, 0)
+	if result, err := b.client.Runs.ReadWithOptions(stopCtx, r.ID, &tfe.RunReadOptions{
+		Include: []tfe.RunIncludeOpt{tfe.RunTaskStages},
+	}); err == nil {
+		taskStages = result.TaskStages
+	} else {
```

Comment:
Although the inline if approach is valid syntax, I don't generally like them spread over multiple lines.

```suggestion
	result, err := b.client.Runs.ReadWithOptions(stopCtx, r.ID, &tfe.RunReadOptions{
		Include: []tfe.RunIncludeOpt{tfe.RunTaskStages},
	})
	if err == nil {
		taskStages = result.TaskStages
	} else {
```

Reply:
Fixed.